api/controllers/v0.2/createPatron.js

```diff
@@ -0,0 +1,193 @@
+ * @param {string} message
+ * @param {string} title
+ * @param {string} debugMessage
+ * @return {object}
+ */
+function collectErrorResponseData(status, type, message, title, debugMessage) {
+  return {
+    status: status || null,
+    type: type || '',
+    message: message || '',
+    title: title || '',
+    debug_message: debugMessage || '',
+  };
+}
+
+/**
+ * renderResponse(req, res, status, message)
+ * Render the response from the ILS API.
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ * @param {number} status
+ * @param {object} message
+ */
+function renderResponse(req, res, status, message) {
+  res
+    .status(status)
+    .header('Content-Type', 'application/json')
+    .json(message);
+}
+
+/**
+ * createPatron(req, res)
+ * The callback for the route "/patrons".
+ * It will fire a POST request to the ILS API for creating a new patron.
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ */
+function createPatron(req, res) {
+  const patronData = req.body;
+  delete patronData.patronCodes.pcode4; // TODO: remove this line once pcode4 is accepted; also remove related line in v0.2 modelStreamPatron.js
+
+  const requiredFields = [];
+
+  if (!patronData || isEmpty(patronData)) {
+    res
+      .status(400)
+      .header('Content-Type', 'application/json')
+      .json(modelResponse.errorResponseData(
+        collectErrorResponseData(
+          null,
+          'invalid-request',
+          'Missing required patron information.',
+          null,
+          { form: ['Patron data appears to be blank.'] } // eslint-disable-line comma-dangle
+        ) // eslint-disable-line comma-dangle
+      ));
+
+    return;
+  }
+
+  // Check if we get all the required information from the client
+  const missingFields = modelDebug.checkMissingRequiredField(requiredFields);
+
+  if (missingFields.length) {
+    const debugMessage = modelDebug.renderMissingFieldDebugMessage(missingFields);
+
+    res
+      .status(400)
+      .header('Content-Type', 'application/json')
+      .json(modelResponse.errorResponseData(
+        collectErrorResponseData(
+          null,
+          'invalid-request',
+          'Missing required patron information.',
+          null,
+          debugMessage // eslint-disable-line comma-dangle
+        ) // eslint-disable-line comma-dangle
+      ));
+
+    return;
+  }
+
+  clientKey = process.env.CACHED_CLIENT_KEY ||
+    awsDecrypt.decryptKMS(process.env.ILS_CLIENT_KEY);
+  clientPassword = process.env.CACHED_CLIENT_PASSWORD ||
+    awsDecrypt.decryptKMS(process.env.ILS_CLIENT_SECRET);
+
+  Promise.all([clientKey, clientPassword]).then((values) => {
+    [process.env.CACHED_CLIENT_KEY, process.env.CACHED_CLIENT_PASSWORD] = values;
+
+    const username = process.env.CACHED_CLIENT_KEY;
+    const password = process.env.CACHED_CLIENT_PASSWORD;
+    const basicAuth = `Basic ${base64(`${username}:${password}`)}`;
+
+    axios.post(process.env.ILS_CREATE_TOKEN_URL, {}, {
```

Comment:
What happens if the `ILS_CREATE_TOKEN_URL` and `ILS_CREATE_PATRON_URL` are both empty for whatever reason? There's no check to make sure they're not empty to then proceed to the API call.